salt/modules/napalm_network.py

```diff
@@ -1547,18 +1547,12 @@ def load_config(filename=None,
 
 
 @salt.utils.napalm.proxy_napalm_wrap
-def load_template(template_name,
+def load_template(template_name=None,
                   template_source=None,
-                  template_hash=None,
```

Comment:
We can't remove any of these older kwargs without putting them on a deprecation path first. We'll need to support these with a warning for 2 feature releases. (So, Sodium would be the version to wan until and then remove the kwargs.) You can use the `warn_until` function in `salt.utils.versions`.

Reply:
I understand that I should revert https://github.com/saltstack/salt/pull/48945/commits/6a70bb1d27a9141ca04d0a3519171c2b074b9588.
In that case, they will no longer be used (although defined in the function header). How would you suggest to proceed? Leave them defined and unused?
Sorry I know this is a bit controversial, it's also the first time I'm deprecating something so I'm not sure what's the best way. Also, please remember that the arguments in subject, as of today, are not used / don't make sense to be there in the first place. Thanks!

Reply:
so, you have `**template_vars`, so you don't want anyone that is still passing stuff to this, to inject extra arguments to the template_vars.

I would put a for loop at the top of the function, that iterates over all of the arguments here, and just tells the user that they do not need to specify them, and in 2 releases, they will be passed to template_vars if they do not stop using them.

That should be good enough, then in 2 releases we will just delete all that, and delete these variables from the function definition.

Reply:
Thanks @gtmanfred - I will update this branch tomorrow.